packages/cdktf-cli/bin/cmds/ui/models/terraform.ts

```diff
@@ -93,10 +93,35 @@ export abstract class AbstractTerraformPlan implements TerraformPlan {
   }
 }
 
+export const getResourceState = (line: string): DeployingResourceApplyState | undefined => {
+  switch (true) {
+    case /Creating.../.test(line):
+    case /Provisioning.../.test(line):
+    case /Still creating.../.test(line):
+    case /Still provisioning.../.test(line):
+      return DeployingResourceApplyState.CREATING;
+    case /Creation complete/.test(line):
+    case /Provisioning complete/.test(line):
+      return DeployingResourceApplyState.CREATED;
+    case /Modifying.../.test(line):
+      return DeployingResourceApplyState.UPDATING;
+    case /Modifications complete/.test(line):
+      return DeployingResourceApplyState.UPDATED;
+    case /Destroying.../.test(line):
+    case /Still destroying.../.test(line):
+      return DeployingResourceApplyState.DESTROYING;
+    case /Destruction complete/.test(line):
+      return DeployingResourceApplyState.DESTROYED;
+    default:
+      return undefined
+  }
+}
+
 export interface Terraform {
   init: () => Promise<void>;
   plan: (destroy: boolean) => Promise<TerraformPlan>;
   deploy(planFile: string, stdout: (chunk: Buffer) => any): Promise<void>;
   destroy(stdout: (chunk: Buffer) => any): Promise<void>;
   output(): Promise<{[key: string]: TerraformOutput}>;
+  outputParser: (line: string) => DeployingResource | undefined;
```

Comment:
I'm not sure this is the best level for this abstraction. I think we'll want the same (or almost the same) parser added to the non-cloud version based on a Terraform version check so that we can use https://www.terraform.io/docs/internals/machine-readable-ui.html

Reply:
I am also not 100% satisfied with this place. Where would you put it? Or should we just always assume the output is json?

Reply:
Left a note in the alternative PR, but probably depends on if the cloud output json is exactly the same format as the cli output json. If so, then probably either always try json and fallback on failure, or just be explicit about which parsing mode (so still at this level, but not the entire parser). If they are different, then this may be the best spot.